### ISSUE TITLE
PresentPaymentSheet: Unhandled Exception: type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?'

### ISSUE THREAD

**Issue by Pablo-Aldana** [flutter-stripe/flutter_stripe#1344]
**Describe the bug**
I've just upgrade to version 9.3.0 and when trying to make a payment when calling to         

await Stripe.instance.presentPaymentSheet();

it does make the payment you can see it on the Stripe dashboard however at the end of the process it does thrown an exception that it is not even captured by the code.
```
VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?'
#0      MethodChannelStripe.presentPaymentSheet
method_channel_stripe.dart:263
<asynchronous suspension>
#1      Stripe.presentPaymentSheet
stripe.dart:517
<asynchronous suspension>
```

**To Reproduce**
1 - Follow all the payment process (payment intent etc..)
2 -  call to presentPaymentSheet

```
 try {
      // 3. display the payment sheet.
    
        await Stripe.instance.presentPaymentSheet();
      checkout();
    } on Exception catch (e) {
      if (e is StripeException) {
        if (e.error.code.name != FailureCode.Canceled.name) {
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(
              content: Text(
                  'Error al hacer el pago con Stripe. Haga captura y contáctenos en : ${e.error.localizedMessage}'),
            ),
          );
        }
      } else {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(
            content: Text(
                'Hubo un error con Stripe. Haga captura y contáctenos en  ${e}'),
          ),
        );
      }
```

It will show the paymentSheet you can make the payment in my example I have tried with Apple Pay and entering manually a card after the payment is done and a successful message is displayed on the screen, the exception is thrown, not capture and stop the process on the app. 

**Expected behavior**
I was expecting the payment to be completed and not exception to be thrown as per version 9.2.2

**Smartphone / tablet**
 - Device: iPhone Simulator
 - OS: 16.4
 - Package version: 9.3.0
 - Flutter version Flutter (Channel stable, 3.10.6, on macOS 13.4.1 22F82 darwin-arm64, locale
    en-GB)

**Additional context**
Probably not related, but right now I'm not even capable to downgrade to 9.2.2 as all the time pubspec.lock pick 9.3.0

**Comment by stenlee** [comment]
I can confirm this issue.

**Comment by habibasseiss** [comment]
This also happens to me.

**Comment by habibasseiss** [comment]
> **Additional context**
> Probably not related, but right now I'm not even capable to downgrade to 9.2.2 as all the time pubspec.lock pick 9.3.0

I could successfully downgrade to 9.2.2 if I force version for all the related packages like this in `pubspec.yaml`:
```
flutter_stripe: 9.2.2
stripe_android: 9.2.1
stripe_ios: 9.2.1
stripe_platform_interface: 9.2.2
```

By doing this, the error for `type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?'` is gone.

**Comment by mjunaidtahir** [comment]
i also faced this issue i  also did this
flutter_stripe: 9.2.2
stripe_android: 9.2.1
stripe_ios: 9.2.1
stripe_platform_interface: 9.2.2
for solution in my project

**Comment by iambenjaminalbertelli** [comment]
It also happens to me.

**Comment by EliaTolin** [comment]
Hello, 

I have fixed this issue in #1333.

Please you try to upgrade with: 

```
  flutter_stripe:
    git:
      url: https://github.com/flutter-stripe/flutter_stripe/
      path: packages/stripe/
      ref: main
```

**Comment by murataslan0** [comment]
> Hello,
> 
> I have fixed this issue in #1333.
> 
> Please you try to upgrade with:
> 
> ```
>   flutter_stripe:
>     git:
>       url: https://github.com/flutter-stripe/flutter_stripe/
>       path: packages/stripe/
>       ref: main
> ```

didn't work for me

**Comment by remonh87** [comment]
Fix is shipped in `stripe_ios: 9.3.1`
